src/Collateral.hs

```diff
@@ -133,7 +133,7 @@ mkValidator contractInfo@ContractInfo{..} dat interestPayDate ctx = validate
     checkDeadline = traceIfFalse "deadline check fail" (contains (from (lendDate dat + repayinterval dat)) (U.range ctx))
 
     checkBorrowerDeadLine :: Bool
-    checkBorrowerDeadLine = traceIfFalse "borrower deadline check fail" (contains (U.range ctx) (from interestPayDate))
+    checkBorrowerDeadLine = after interestPayDate (U.range ctx)
```

Comment:
Actually you could the same thing you did in `Request.hs`, where `getUpperBound` is used as `lendDate`: use `interestPayDate = getUpperBound $ U.range ctx` (and thus also get rid of the redeemer).

However, it's up to you, this works too.